### ISSUE TITLE
How does shuffle = 'batch' work in the background?

### ISSUE THREAD

**Issue by tryfonch** [fchollet/keras#7560]
When I train the model using the .fit() layer there is the argument **shuffle** preset to **True**.

Let's say that my dataset has 100 samples and that the batch size is 10. When I set **shuffle = True** then keras first randomly selects randomly the samples (now the 100 samples have a different order) and on the new order it will start creating the batches: batch 1: 1-10, batch 2: 11-20 etc. 

If I set **shuffle = 'batch'** how is it supposed to work in the background? Intuitively and using the previous example of 100 samples dataset with batch size = 10 my guess would be that keras first allocates the samples to the batches (i.e. batch 1: samples 1-10 following the dataset original order, batch 2: 11-20 following the dataset original order as well, batch 3 ... so on so forth) and then shuffles the order of the batches. So the model now will be trained on the randomly ordered batches say for example: 3 (contains samples 21 - 30), 4 (contains samples 31 - 40), 7 (contains samples 61 - 70), 1 (contains samples 1 - 10), ... (I made up the order of the batches).

Is my thinking right or am I missing something?

Thanks!

**Comment by AdityaSoni19031997** [comment]
shuffle: boolean, whether to shuffle the training data before each epoch.

**Comment by tryfonch** [comment]
@AdityaSoni19031997  shuffle except for **True**, **False** can also take **batch** value according to their documentation. In that case how does the shuffling work?

**Comment by hoonkai** [comment]
I think the doc is just wrong. `shuffle = "batch"` isn't implemented.

**Comment by m-schroeck** [comment]
Not true, it is implemented. See keras/keras/engine/training_arrays.py

**Comment by cyril-p** [comment]
Same questionÇÇ

**Comment by mcgibbon** [comment]
From the [documentation](https://keras.io/models/sequential/):

> batch' is a special option for dealing with the limitations of HDF5 data; it shuffles in batch-sized chunks.
